### PR TITLE
Add `FiltrationProcess` class and supporting slots and enumerations

### PR DESCRIPTION
**Pull request by JamesTessmer**
PR to add filtration and phase separation classes and supporting slots.

### CODE REVIEW

**Comment by turbomam** [comment]
Thanks for the new commit @JamesTessmer. Will you be updating the PR to address the error form the GH actions?

**Comment by JamesTessmer** [comment]
> Thanks for the new commit @JamesTessmer. Will you be updating the PR to address the [error from the GH actions](https://github.com/microbiomedata/nmdc-schema/actions/runs/6855108387/job/18639473019?pr=1326#step:7:317)?

I am planning on fixing that later today, unfortunately i had some issues pop up today because of a power surge in our data center over the weekend so I won’t be able to touch this up until later today or early tomorrow.

**Comment by JamesTessmer** [comment]
Build currently fails, but it should pass when main is merged in. Once any remaining build issues are addressed this should be ready for review.

**Comment by JamesTessmer** [comment]
Merged main in, resolved all the conflicts. This should be ready for review.

**Review by turbomam — CHANGES_REQUESTED** [review]
Are any of the people contributing to this PR generating the documentation locally with `make testdoc`? That can be a really helpful for exploring the relationships between all of the terms added in a PR.

- What does it mean that a `FiltrationProcess` can have a `separation_method` relationship with a `ContainedInEnum`?
- What does it mean that a `FiltrationProcess` can have a `contained_in` relationship with a `ContainedInEnum`? I appreciate effort to use a verb-like slot name, but I really doubt you mean to say the `FiltrationProcess` is `contained_in` in something. Maybe something like `sends_output_into` would make more sense.
- Let's think about using the `ContainedInEnum` for other purposes. It really seems more like a ContainerCategoryEnum to me.
- What is a `MaterialContainer`? I may be the person who created that term, so I am willing to writ e description, but I will want your review. I think my intention here was to differentiate it from a data container, like the `Database` class.
- How can a `MaterialContainer` be `contained_in` in something? Maybe I have put too much pressure on you to be parsimonious with new slots. I think we need to say that
  - a `FiltrationProcess` `sends_output_into` some `ContainerCategoryEnum`
  - a `MaterialContainer` `container_category`  some `ContainerCategoryEnum`
- Are you sure we want to use strings (and not enumerations) for
  - `conditionings`
  - `filter_material`
  - `filtration_category`
- what can "either phase" in the `conditionings` slot `description` be? I assume solid, liquid or gas. Slot descriptions shouldn't require anyone to guess.
- we should add [list_elements_ordered](https://linkml.io/linkml-model/latest/docs/list_elements_ordered/)  to `conditionings`
- `instrument_used` needs a `description`. Do you want to do that or should I?
- `material` seems like an orphaned slot. Does anybody in metabolink remember what it was for?

**Review by turbomam — CHANGES_REQUESTED** [review]

**Review by turbomam — COMMENTED** [review]
I have updated the title of this PR so that it will be more informative in the auto-generated release notes.

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -2153,16 +2200,39 @@ enums:
...
       - ORCID:0009-0001-1555-1601 #Anastasiya Prymolenna
       - ORCID:0000-0002-8683-0050 #Montana Smith
     permissible_values:
       V-bottom conical tube:
+      PTFE 96-well filter plate:
 
-slots:
+  ConditioningSolventEnum:
+    description: The solvents that may be used in processing metabolomic samples.
+    permissible_values:
```
[the remark this answers is not in the document]
Comment (anastasiyaprymolenna) [reply src/schema/nmdc.yaml:2214]: @mslarae13 Can you comment on how proportions should be indicated?

File: src/schema/nmdc.yaml [hunk 2]
```diff
@@ -2153,16 +2200,39 @@ enums:
...
     description: The permitted types of containers used in processing metabolomic samples.
     contributors:
       - ORCID:0009-0001-1555-1601 #Anastasiya Prymolenna
       - ORCID:0000-0002-8683-0050 #Montana Smith
     permissible_values:
       V-bottom conical tube:
+      PTFE 96-well filter plate:
 
-slots:
+  ConditioningSolventEnum:
```
[the remark this answers is not in the document]
Comment (anastasiyaprymolenna) [reply src/schema/nmdc.yaml:2212]: This has been removed. It was missed when we removed the ConditioningValue class

File: src/schema/basic_slots.yaml [hunk 3]
```diff
@@ -121,12 +121,21 @@ slots:
     todos:
       - add date string validation pattern
 
+  instrument_used:
+    comments:
+      - not to be confused with 'used' which currently has a string range
+    todos:
+      - annotate me!
```
[the remark this answers is not in the document]
Comment (JamesTessmer) [reply src/schema/basic_slots.yaml:128]: Added "The device or tool that performed or aided in performing a process."

File: src/schema/nmdc.yaml [hunk 4]
```diff
@@ -2849,8 +2919,8 @@ slots:
       - ORCID:0009-0001-1555-1601 #Anastasiya Prymolenna
       - ORCID:0000-0002-8683-0050 #Montana Smith
 
-  container_type:
-    range: ContainerTypeEnum
+  contained_in:
```
[the remark this answers is not in the document]
Comment (JamesTessmer) [reply src/schema/nmdc.yaml:2922]: Updated the ContainedInEnum to be a ContainerCategory

**Comment by anastasiyaprymolenna** [comment]
> Are any of the people contributing to this PR generating the documentation locally with `make testdoc`? That can be a really helpful for exploring the relationships between all of the terms added in a PR.
> 
> * What does it mean that a `FiltrationProcess` can have a `separation_method` relationship with a `ContainedInEnum`?
> * What does it mean that a `FiltrationProcess` can have a `contained_in` relationship with a `ContainedInEnum`? I appreciate the effort to use a verb-like slot name, but I really doubt you mean to say the `FiltrationProcess` is `contained_in` in something. Maybe something like `sends_output_into` would make more sense.

This stems from the fact that the way we have written the enum for Containers is very ambiguous so any slot that uses a container as it's range (meaning a process uses a container) it will have this relationship.

> * Let's think about using the `ContainedInEnum` for other future purposes. It really seems more like a ContainerCategoryEnum to me.

This has been updated accordingly.

> * What is a `MaterialContainer`? I may be the person who created that term, so I am willing to write a description, but I will want your review. I think my intention here was to differentiate it from a data container, like the `Database` class.
> * How can a `MaterialContainer` be `contained_in` in something? Maybe I have put too much pressure on you to be parsimonious with new slots. I think we need to say that
>   
>   * a `FiltrationProcess` `sends_output_into` some `ContainerCategoryEnum`
>   * a `MaterialContainer` `container_category`  some `ContainerCategoryEnum`


This comment relates to `sample_prep.yaml`. This file has existed since before MetaboLink started. We do not use anything here including `MaterialContainer`. We only comment out conflicts from here before a commit. We do not add to this file.

> * Are you sure we want to use strings (and not enumerations) for
>   
>   * `conditionings`
>   * `filter_material`
>   * `filtration_category`

These are not slots that inform any downstream processing, so its ok to not enforce Enums. It is more important to just note what was done in the details but not needed to filter down by a consistent value. Someone would want to be able to reference this info, but not query on it.

> * what can "either phase" in the `conditionings` slot `description` be? I assume solid, liquid or gas. Slot descriptions shouldn't require anyone to guess.

@mslarae13 this is a description we pulled from an ontology. I hesitate to change the language because it is specific to the reference (which we do not list in the description, but it is quoted)

> * we should add [list_elements_ordered](https://linkml.io/linkml-model/latest/docs/list_elements_ordered/)  to `conditionings`

Done

> * `instrument_used` needs a `description`. Do you want to do that or should I?

Done

> * `material` seems like an orphaned slot. Does anybody in metabolink remember what it was for?

Yes this is removed for this merge. It was prep for PhaseSeparation class.

**Review by turbomam — APPROVED** [review]
thanks everybody for the valuable discussions. I may raise some new issues about this, but I'm happy to merge this in now.

**Pull request merged**